src/System.CommandLine/Command.cs

```diff
@@ -129,59 +166,49 @@ public void AddGlobalOption(Option option)
 
         internal Parser? ImplicitParser { get; set; }
 
-        private protected void AddSymbol(Symbol symbol)
-        {
-            Children.AddWithoutAliasCollisionCheck(symbol);
-            symbol.AddParent(this);
-        }
-
         /// <inheritdoc />
         public override IEnumerable<CompletionItem> GetCompletions(CompletionContext context)
         {
-            if (Children.Count == 0)
+            List<CompletionItem> completions = new();
+
+            foreach (Command command in SubCommands)
```

Comment:
I made some decent performance improvements earlier this year by using `for` instead of `foreach`. Should we be avoiding `IEnumerable<T>` and `foreach` when `IReadOnlyList<T>` is available?